usaspending_api/etl/management/commands/swap_in_new_table.py

```diff
@@ -0,0 +1,292 @@
+    db_table_name: str
+    temp_db_table_name: str
+
+    # Query values are populated as they are run during validation and saved for re-use
+    query_result_lookup = {
+        "temp_table_constraints": [],
+        "curr_table_constraints": [],
+        "temp_table_indexes": [],
+        "curr_table_indexes": [],
+    }
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "--table",
+            type=str,
+            help="The Postgres Table to swap with another containing the same name with '_temp' appended",
+        )
+        parser.add_argument(
+            "--keep-old-data",
+            action="store_true",
+            default=False,
+            help="Indicates whether or not to drop old table at the end of the command",
+        )
+        parser.add_argument(
+            "--allow-foreign-key",
+            action="store_true",
+            default=False,
+            help="Indicates whether Foreign Key constraints should be allowed."
+            " It is advised to not allow Foreign Key constraints since they can cause deadlock.",
+        )
+
+    def handle(self, *args, **options):
+        self.db_table_name = options["table"]
+        self.temp_db_table_name = f"{self.db_table_name}_temp"
+
+        with connection.cursor() as cursor:
+            self.validate_state_of_tables(cursor, options)
+            self.swap_index_sql(cursor)
+            self.swap_constraints_sql(cursor)
+            self.swap_table_sql(cursor)
+            if not options["keep_old_data"]:
+                self.drop_old_table_sql(cursor)
+            self.extra_sql(cursor)
+
+    def swap_index_sql(self, cursor):
+        logging.info("Renaming indexes of the new and old tables.")
+        temp_indexes = self.query_result_lookup["temp_table_indexes"]
+        curr_indexes = self.query_result_lookup["curr_table_indexes"]
+        rename_sql = []
+        sql_template = "ALTER INDEX {old_index_name} RENAME TO {new_index_name};"
+        for val in curr_indexes:
+            old_name = val["indexname"]
+            new_name = f"{old_name}_old"
+            rename_sql.append(sql_template.format(old_index_name=old_name, new_index_name=new_name))
+        for val in temp_indexes:
+            old_name = val["indexname"]
+            new_name = re.match("^(.*)_temp$", old_name, flags=re.I)[1]
+            rename_sql.append(sql_template.format(old_index_name=old_name, new_index_name=new_name))
+
+        if rename_sql:
+            cursor.execute("\n".join(rename_sql))
+
+    def swap_constraints_sql(self, cursor):
+        logging.info("Renaming constraints of the new and old tables.")
+        temp_constraints = self.query_result_lookup["temp_table_constraints"]
+        curr_constraints = self.query_result_lookup["curr_table_constraints"]
+        rename_sql = []
+        sql_template = "ALTER TABLE {table_name} RENAME CONSTRAINT {old_constraint_name} TO {new_constraint_name};"
```

Comment:
Are you sure these don't need to be qualified by the schema name fo the table, if for instance the user does not have permission to see ALL schemas?

Reply:
Correct. It checks that both the active table and the active table with `_temp` appended exist prior to moving forward with other checks.